catalog/core/catalog-core-api-impl/src/test/java/ddf/catalog/util/impl/QueryResultIteratorTest.java

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.util.impl;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ddf.catalog.data.Result;
+import ddf.catalog.util.impl.QueryResultIterable.QueryResultIterator;
+
+public class QueryResultIteratorTest {
+
+    private QueryResultPaginator paginatorMock;
+
+    private QueryResultIterator queryIterator;
+
+    private Iterator<Result> queriedResultsMock;
+
+    private List<Result> resultListMock;
+
+    private Result resultMock;
+
+    @Before
+    public void setUp() throws Exception {
+
+        paginatorMock = mock(QueryResultPaginator.class);
+        queriedResultsMock = mock(Iterator.class);
+        resultListMock = mock(List.class);
+        resultMock = mock(Result.class);
+
+        queryIterator = new QueryResultIterator(paginatorMock);
+    }
+
+    @Test
+    public void testNextResultsExistExpectResult() {
+
+        boolean isResultsListEmpty = false;
+
+        when(queriedResultsMock.hasNext()).thenReturn(!isResultsListEmpty);
+        when(queriedResultsMock.next()).thenReturn(resultMock);
+        queryIterator.setQueriedResults(queriedResultsMock);
```

Comment:
❗️ Instead of setting this here, we probably want to mock the call to the paginator and look at the sequence of calls from the iterator and paginator point of view.